azure_iot_edge/README.md

```diff
@@ -0,0 +1,146 @@
+
+1. Edit your IoT Edge device `config.yaml` file:
+    - **Linux**: Make sure the `connect.management_uri` and `listen.management_uri` options point to a Unix Domain Socket (note that this is the default configuration). For example:
+
+        ```yaml
+        # /etc/iotedge/config.yaml
+
+        connect:
+            management_uri: "unix:///var/run/iotedge/mgmt.sock"
+            # ...
+
+        listen:
+            management_uri: "fd://iotedge.mgmt.socket"
+            # ...
+        ```
+
+    - **Windows**: Make sure the `connect.management_uri` and `listen.management_uri` options point to an HTTP endpoint. For example:
+
+        ```yaml
+        # /etc/iotedge/config.yaml
+
+        connect:
+            management_uri: "http://localhost:15580"
+            # ...
+
+        listen:
+            management_uri: "http://localhost:15580"
+            # ...
+        ```
+
+1. Configure the **Edge Agent** runtime module as follows:
+    - Image version must be `1.0.10` or above.
+    - Under "Environment Variables", experimental metrics must be enabled by adding these environment variables (note the double underscores):
+        - `ExperimentalFeatures__Enabled`: `true`
+        - `ExperimentalFeatures__EnableMetrics`: `true`
+
+1. Configure the **Edge Hub** runtime module as follows:
+    - Image version must be `1.0.10` or above.
+    - Under "Environment Variables", experimental metrics must be enabled by adding these environment variables (note the double underscores):
+        - `ExperimentalFeatures__Enabled`: `true`
+        - `ExperimentalFeatures__EnableMetrics`: `true`
+
+1. Install and configure the Datadog Agent as a **custom module**:
+    - Set the module name. For example: `datadog-agent`.
+    - Set the Agent image URI. For example: `datadog/agent:7.24.0`.
+    - Under "Environment Variables", configure your `DD_API_KEY`. You may also set extra Agent configuration here (see [Agent Environment Variables][4]).
+    - Under "Container Create Options", enter the following configuration based on your device OS. Edit the `com.datadoghq.ad.instances` option as appropriate. See the [sample azure_iot_edge.d/conf.yaml][5] for all available configuration options. See the documentation on [Docker Integrations Autodiscovery][6] for more information on labels-based integration configuration. **Note**: `NetworkId` must correspond to the network name set in the device `config.yaml` file.
+        - Linux:
+            ```json
+            {
+                "HostConfig": {
+                    "NetworkMode": "default",
+                    "Env": ["NetworkId=azure-iot-edge"],
+                    "Binds": [
+                        "/var/run/docker.sock:/var/run/docker.sock",
+                        "/var/run/iotedge/mgmt.sock:/var/run/iotedge/mgmt.sock"
+                    ]
+                },
+                "Labels": {
+                    "com.datadoghq.ad.check_names": "[\"azure_iot_edge\"]",
```

Comment:
Right. I went off from https://docs.datadoghq.com/agent/docker/integrations/ and applied the approach there to the Datadog Agent container itself, but I'd agree that in retrospect it's a bit "exotic".

I could add these labels to an arbitrary container, like one of Edge Hub or Edge Agent (not both, since we only want a single instance).

Mounting a config file is a bit tricky (need to deal with mount paths and making sure the Agent is well configured), I think since we're using the Docker Agent using autodiscovery is preferable. Another con to mounting a file is that device operators will most likely define everything through the Azure IoT Edge web UI, and there's no way to define arbitrary files there (not AFAIK at least). Defining labels in the Create Options should sound more familiar to device operators.

Reply:
Okay, moved these labels to the Edge Agent container :+1: